financial-connections/src/main/java/com/stripe/android/financialconnections/domain/GetOrFetchSync.kt

```diff
@@ -17,10 +17,44 @@ internal class GetOrFetchSync @Inject constructor(
     @Named(APPLICATION_ID) private val applicationId: String,
 ) {
 
-    suspend operator fun invoke(): SynchronizeSessionResponse {
-        return repository.getOrFetchSynchronizeFinancialConnectionsSession(
+    suspend operator fun invoke(
+        fetchCondition: FetchCondition = FetchCondition.IfMissing
+    ): SynchronizeSessionResponse {
+        return repository.getOrSynchronizeFinancialConnectionsSession(
             clientSecret = configuration.financialConnectionsSessionClientSecret,
-            applicationId = applicationId
+            applicationId = applicationId,
+            fetchCondition = fetchCondition::check,
         )
     }
+
+    sealed interface FetchCondition {
+        fun check(response: SynchronizeSessionResponse): Boolean
+
+        /**
+         * Session won't be fetched if it's already cached.
+         */
+        data object IfMissing : FetchCondition {
+            override fun check(response: SynchronizeSessionResponse): Boolean {
+                return false
+            }
+        }
```

Comment:
Nit: Should we rename this `NoFetch` or something along those lines? The case itself doesn’t have any “reload if missing” logic in it.